### PR TITLE
[Components] statuspage #5331

### PR DESCRIPTION
**Pull request by lcaresia**

### CODE REVIEW

**Comment by andrewjschuang** [comment]
Resolves #5380.

File: components/statuspage/actions/create-incident/create-incident.mjs [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+      label: "Name",
+      description: "The name of the incident",
+      type: "string",
+    },
+    status: {
+      label: "Status",
+      description: "The status of the incident. (e.g. `resolved`, `investigating`)",
+      type: "string",
+      options: constants.INCIDENT_STATUSES,
+    },
```
Comment (andrewjschuang) [inline components/statuspage/actions/create-incident/create-incident.mjs:28]: Can this prop be in the app file?

File: components/statuspage/actions/update-incident/update-incident.mjs [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+          pageId: c.pageId,
+        }),
+      ],
+    },
+    status: {
+      label: "Status",
+      description: "The new status of the incident (e.g. `resolved`, `investigating`)",
+      type: "string",
+      options: constants.INCIDENT_STATUSES,
+    },
```
Comment (andrewjschuang) [inline components/statuspage/actions/update-incident/update-incident.mjs:32]: Can this prop be in the app file?

File: components/statuspage/sources/new-incident-updated/new-incident-updated.mjs [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+          limit: 20,
+        },
+      });
+
+      incidents.reverse().forEach(this.emitEvent);
+    },
+    emitEvent(event) {
+      const data = event?.incident ?? event;
+
+      if (data.incident_updates.length <= 1) {
```
Comment (andrewjschuang) [inline components/statuspage/sources/new-incident-updated/new-incident-updated.mjs]: Is there are reason we skip when there is one incident only?

File: components/statuspage/sources/incident-updated/incident-updated.mjs [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+          limit: 20,
+        },
+      });
+
+      incidents.reverse().forEach(this.emitEvent);
+    },
+    emitEvent(event) {
+      const data = event?.incident ?? event;
+
+      if (data.incident_updates.length <= 1) {
```
Comment (andrewjschuang) [inline components/statuspage/sources/incident-updated/incident-updated.mjs:26]: Is there a reason we skip when there is only one?

**Review by andrewjschuang — APPROVED** [review]
LGTM!
I left just a few questions below

File: components/statuspage/sources/incident-updated/incident-updated.mjs (hunk 4 above)
Comment (lcaresia) [reply components/statuspage/sources/incident-updated/incident-updated.mjs:26]: Yes, because if just have one it's a new incident, not an updated.

**Review by andrewjschuang — APPROVED** [review]

**Pull request merged**